File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.c
int findFinalValue(int* nums, int numsSize, int original) {
    int target = original;
    bool looking = true;
    while (looking){
        for(int i = 0; i < numsSize; i++){
            if (i == numsSize - 1 && nums[i] != target){
                looking = false;
                break;
            }
            if (nums[i] == target){
                target = 2 * target;
                break;
            }
        }
    }
    return target;
}

File: 2274-keep-multiplying-found-values-by-two/keep-multiplying-found-values-by-two.js
/**
 * @param {number[]} nums
 * @param {number} original
 * @return {number}
 */
var findFinalValue = function(nums, original) {
    let numSet = new Set(nums);
    let present = numSet.has(original);
    while (present){
        original*=2;
        present = numSet.has(original);
    }
    return original;
};